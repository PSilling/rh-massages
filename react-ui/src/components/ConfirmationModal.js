// react imports
import React, { Component } from 'react';

// component imports
import ModalActions from './ModalActions';

// module imports
import {ModalContainer, ModalDialog} from 'react-modal-dialog';

// util imports
import _t from '../utils/Translations.js';

/**
 * Custom confirmation modal component.
 */
class ConfirmationModal extends Component {

  render() {
    return(
      <ModalContainer onClose={this.props.onClose}>
        <ModalDialog onClose={this.props.onClose} width="40%">
            <h3>{ _t.translate('Action confirmation') }</h3>
            <p>{this.props.message}</p>
            <ModalActions
              primaryLabel={ _t.translate('Proceed') }
              onProceed={this.props.onConfirm}
              onClose={this.props.onClose}
            />
          </ModalDialog>
      </ModalContainer>
    )
  }
}

export default ConfirmationModal
