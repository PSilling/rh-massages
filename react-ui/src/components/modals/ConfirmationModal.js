// react imports
import React, { Component } from "react";
import ReactDOM from "react-dom";
import PropTypes from "prop-types";

// module imports
import { ModalContainer, ModalDialog } from "react-modal-dialog";

// component imports
import ModalActions from "../buttons/ModalActions";

// util imports
import _t from "../../util/Translations";

/**
 * Modal with a simple action confirmation message and ModalActions.
 */
class ConfirmationModal extends Component {
  handleKeyPress = event => {
    if (event.charCode === 13 && document.activeElement === ReactDOM.findDOMNode(this.modal)) {
      this.props.onConfirm();
    }
  };

  renderInsides = () => (
    <div>
      <h3>{_t.translate("Action confirmation")}</h3>
      <p>{this.props.message}</p>
      <ModalActions onProceed={this.props.onConfirm} onClose={this.props.onClose} autoFocus>
        {this.props.children}
      </ModalActions>
    </div>
  );

  render() {
    return this.props.withPortal ? (
      <ModalContainer onClose={this.props.onClose}>
        <ModalDialog
          onClose={this.props.onClose}
          width="40%"
          style={{ outline: "none" }}
          tabIndex="-1"
          onKeyPress={this.handleKeyPress}
          ref={dialog => {
            this.modalDialog = dialog;
          }}
        >
          {this.renderInsides()}
        </ModalDialog>
      </ModalContainer>
    ) : (
      this.renderInsides()
    );
  }
}

ConfirmationModal.propTypes = {
  /** children supplied to ModalActions */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  /** message in the Modal */
  message: PropTypes.string.isRequired,
  /** callback function triggered on Modal close */
  onClose: PropTypes.func.isRequired,
  /** callback function triggered on primary button click */
  onConfirm: PropTypes.func.isRequired,
  /** whether ModalContainer should be used; useful for testing to avoid portals */
  withPortal: PropTypes.bool
};

ConfirmationModal.defaultProps = {
  children: null,
  withPortal: true
};

export default ConfirmationModal;
