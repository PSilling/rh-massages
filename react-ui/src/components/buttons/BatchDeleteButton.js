// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// component imports
import ConfirmationModal from "../modals/ConfirmationModal";

// util imports
import _t from "../../util/Translations";

/**
 * A button used for batch Massage deletion. Contains a ConfirmationModal to confirm
 * the action.
 */
class BatchDeleteButton extends Component {
  state = { active: false };

  handleToggle = () => {
    this.setState(prevState => ({ active: !prevState.active }));
  };

  render() {
    return (
      <span>
        <button type="button" className="btn btn-default" onClick={this.handleToggle} disabled={this.props.disabled}>
          {this.props.label}
        </button>
        {this.state.active && (
          <ConfirmationModal
            message={_t.translate("Are you sure? This action cannot be reverted.")}
            onClose={this.handleToggle}
            onConfirm={() => {
              this.handleToggle();
              this.props.onDelete();
            }}
          />
        )}
      </span>
    );
  }
}

BatchDeleteButton.propTypes = {
  /** function to be called on action confirmation */
  onDelete: PropTypes.func.isRequired,
  /** whether the button should be disabled */
  disabled: PropTypes.bool,
  /** button label to be displayed */
  label: PropTypes.string
};

BatchDeleteButton.defaultProps = {
  disabled: false,
  label: "Delete"
};

export default BatchDeleteButton;
