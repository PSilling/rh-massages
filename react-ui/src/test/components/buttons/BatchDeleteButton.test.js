// react imports
import React from 'react';
import TestRenderer from 'react-test-renderer';

// test imports
import BatchDeleteButton from '../../../components/buttons/BatchDeleteButton';
import ConfirmationModal from '../../../components/modals/ConfirmationModal';

test('renders content with correct props', () => {
  const testFunction = jest.fn(),
        testRenderer = TestRenderer.create(
          <BatchDeleteButton onDelete={testFunction} label="test" />
        ),
        testInstance = testRenderer.root;

  let button = testInstance.findByType('button');

  button.props.onClick();

  let dialog = testInstance.findByType(ConfirmationModal),
      treeJSON = testRenderer.toJSON();

  expect(testFunction).not.toHaveBeenCalled();

  dialog.props.onConfirm();

  expect(testInstance.instance.state.active).toBe(false);
  expect(testFunction).toHaveBeenCalledTimes(1);
  expect(button.props.children).toEqual("test");
  expect(button.props.disabled).toBe(false);
  expect(treeJSON).toMatchSnapshot();
});
