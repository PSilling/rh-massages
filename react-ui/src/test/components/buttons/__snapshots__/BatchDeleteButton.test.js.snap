// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
<span>
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    test
  </button>
</span>
`;
